### PR TITLE
Subcodes

### PR DESCRIPTION
**Pull request by RalphBln**

### CODE REVIEW

File: schemas/top-api.yml [hunk 1]
```diff
@@ -1671,6 +1669,12 @@ paths:
           schema:
             type: string
           example: snomed
+        - name: subtree
```
Comment (ChristophB) [inline schemas/top-api.yml:1672]: Why introduce another parameter name here? I would stick to 'scope' or 'codeScope'.

File: schemas/top-api.yml [hunk 2]
```diff
@@ -2728,8 +2732,10 @@ components:
         code:
           type: string
           example: 3137-7
-        scope:
-          $ref: '#/components/schemas/CodeScope'
+        children:
```
Comment (ChristophB) [inline schemas/top-api.yml:2735]: I need to check the implementations in the other components before deciding on this change. Originally it was planned for the backend to collect sub codes. By replacing `scope` with `children`, the work now has to be done by the frontend, resulting in a potentially large number of sub codes.

File: schemas/top-api.yml [hunk 3]
```diff
@@ -1671,6 +1669,12 @@ paths:
           schema:
             type: string
           example: snomed
+        - name: subtree
+          in: query
+          description: Indicates whether only the code (self) or its child codes (subtree/leaves) should be returned. Defaults to "self".
```
Comment (ChristophB) [inline schemas/top-api.yml:1674]:
```suggestion
          description: Indicates whether only the code (self) or its child codes (subtree/leaves) should be returned. Defaults to "SELF".
```

**Review by ChristophB — CHANGES_REQUESTED** [review]
As mentioned in on of the comments, I need to check the implementations in the other components.

**Comment by ChristophB** [comment]
> As mentioned in one of the comments, I need to check the implementations in the other components.

So it looks like sub codes are collected by the backend on code lookup requests. They are then appended to the JSON and sent back to the backend for storage.

There is one place in the frontend, where the `scope` property is still referenced. But this pull request removes it completely.
https://github.com/Onto-Med/top-frontend/pull/348/files#diff-5cad3941000b5cd3e9a4e64737429d1ee2b5fecf2aa3a8e0c0811a0898917127R153

File: schemas/top-api.yml (hunk 2 above)
Comment (RalphBln) [reply schemas/top-api.yml:2735]: The codes **are** collected by / stored in the backend. They need to be presented in the frontend, though, as it was planned to give users the ability to manually edit the subtrees (i.e., delete unneeded branches/leaves).

File: schemas/top-api.yml (hunk 1 above)
Comment (RalphBln) [reply schemas/top-api.yml]: Resolved by 898496346ceb7577233d44771010fced11a04a6b

**Comment by RalphBln** [comment]
The top-frontend and top-backend are free of references to Code.scope now. Idk if one of the extensions references it though.

**Review by ChristophB — APPROVED** [review]

**Pull request merged**